src/ucp/core/ucp_ep.c

```diff
@@ -1143,10 +1144,38 @@ static void ucp_ep_set_lanes_failed(ucp_ep_h ep, uct_ep_h *uct_eps)
     }
 }
 
+static int ucp_ep_can_unprogress_lane(ucp_ep_h ep, uct_ep_h uct_ep)
```

Comment:
IMO better to pass lane index instead of uct_ep to guarantee that uct_ep is a lane of this particular UCP ep

Reply:
still stick to uct_ep instead of lane because I re-organize codes following Yossi's guide